### PR TITLE
Add styled content while the page is loading

### PR DESCRIPTION
**Pull request by TimvdLippe**
This adds temporary content while the page is loading. There is an instant response of the actual page, while all elements are downloaded from the server. Once the client finishes the request and stamps out the content, the replacement is hidden and the actual site is shown.

Run `npm update` to install the dependency `polyclean`.

### CODE REVIEW

File: app/index.html [hunk 1]
```diff
@@ -52,14 +72,15 @@
...
-  <link rel="import" href="elements/elements.html" async>
-  <!-- endreplace-->
 </head>
-
 <body class="fullbleed layout vertical">
 <span id="browser-sync-binding"></span>
+  <div id="replacement-content">
+    <div class="header"></div>
+    <div class="content">
+      <span>AREA51LAN</span>
```
Comment (skamoen) [inline app/index.html]: Adten.

File: gulpfile.js [hunk 2]
```diff
@@ -177,6 +178,13 @@ gulp.task('vulcanize', function() {
       inlineCss: true,
       inlineScripts: true
     }))
+    .pipe(polyclean.uglifyJs())
```
Comment (TimvdLippe) [inline gulpfile.js]: After some thinking this should probably be just `polybuild()`, have to test it out

File: app/scripts/app.js [hunk 3]
```diff
@@ -9,18 +9,25 @@ subject to an additional IP rights grant found at http://polymer.github.io/PATEN
...
   };
+
+  var link = document.createElement('link');
+  link.setAttribute('rel', 'import');
+  link.setAttribute('href', 'elements/elements.html');
+  link.setAttribute('async', 'true');
+  document.head.appendChild(link);
+
+  app.addEventListener('dom-change', function() {
+    document.querySelector('#replacement-content').style.display = 'none';
```
Comment (TimvdLippe) [inline app/scripts/app.js]: Replace with `getElementById` (http://stackoverflow.com/questions/15046356/performance-using-js-queryselector)

File: app/scripts/app.js [hunk 4]
```diff
@@ -9,18 +9,25 @@ subject to an additional IP rights grant found at http://polymer.github.io/PATEN
...
-        drawerPanel.closeDrawer();
-      }
-    };
+  // Grab a reference to our auto-binding template
+  // and give it some initial binding values
+  // Learn more about auto-binding templates at http://goo.gl/Dx1u2g
+  var app = document.querySelector('#app');
+  // Close drawer after menu item is selected if drawerPanel is narrow
+  app.onMenuSelect = function() {
+    var drawerPanel = document.querySelector('#paperDrawerPanel');
```
Comment (TimvdLippe) [inline app/scripts/app.js]: Replace with `getElementById` (http://stackoverflow.com/questions/15046356/performance-using-js-queryselector)

File: gulpfile.js [hunk 5]
```diff
@@ -290,7 +299,7 @@ gulp.task('default', ['clean'], function(cb) {
     ['copy', 'styles'],
     'elements',
     ['images', 'fonts', 'html'],
-    'build',
+    'vulcanize', 'build',
```
Comment (TimvdLippe) [inline gulpfile.js]: `vulcanize` does not reflect the behavior of the task, as it now crushes `elements.html`. Probably should take a look at the naming.

File: app/index.html [hunk 6]
```diff
@@ -42,6 +42,26 @@
   <!-- Tile icon for Win8 (144x144) -->
   <meta name="msapplication-TileImage" content="images/touch/ms-touch-icon-144x144-precomposed.png">
 
+  <!-- Temporary styling for slow loading pages -->
+  <style>
+    #replacement-content .header {
```
Comment (TimvdLippe) [inline app/index.html]: Maybe fading out replacementcontent and fading in the paperheaderpanel reduces the flash of content

**Comment by TimvdLippe** [comment]
@svenpopping Could you style the loading element?

**Comment by TimvdLippe** [comment]
@skamoen did you have any objections on these changes?

**Comment by skamoen** [comment]
I think the impact on devices that don't need this is too big for the small gain we would have on devices that do. I haven't properly checked this though.

**Comment by skamoen** [comment]
Let's re-evaluate after implementing lazy-loading next year.

**Pull request closed without merging**